src/components/application_manager/test/commands/hmi/get_urls_test.cc

```diff
@@ -81,237 +81,255 @@ const std::string kInitialService = "0x0";
 const std::string kPolicyService = "7";
 const std::string kDefaultUrl = "URL is not found";
 const std::string kDefaultId = "default";
+const std::string kPolicyAppId = "policy_app_id";
 }  // namespace
 
-class GetUrlsTest : public CommandsTest<CommandsTestMocks::kIsNice> {
+class GetUrlsTest : public CommandRequestTest<CommandsTestMocks::kIsNice> {
  public:
-  policy_test::MockPolicyHandlerInterface policy_handler_;
-  utils::SharedPtr<policy_manager_test::MockPolicyManager> mock_policy_manager_;
-  NiceMock<event_engine_test::MockEventDispatcher> mock_event_dispatcher_;
-  ApplicationSharedPtr app;
+  policy_test::MockPolicyHandlerInterface mock_policy_handler_;
+  MessageSharedPtr command_msg_;
+  RequestFromHMIPtr request_command_;
+
+  GetUrlsTest() {
+    command_msg_ =
+        CreateMessage(NsSmartDeviceLink::NsSmartObjects::SmartType_Map);
+    (*command_msg_)[am::strings::params][am::strings::connection_key] =
+        kConnectionKey;
+    (*command_msg_)[am::strings::msg_params][am::hmi_request::service] =
+        kInitialService;
+
+    request_command_ = CreateCommand<GetUrls>(command_msg_);
 
-  virtual void SetUp() OVERRIDE {
     ON_CALL(mock_app_manager_, GetPolicyHandler())
-        .WillByDefault(ReturnRef(policy_handler_));
-    mock_policy_manager_ =
-        utils::MakeShared<policy_manager_test::MockPolicyManager>();
-    ASSERT_TRUE(mock_policy_manager_);
+        .WillByDefault(ReturnRef(mock_policy_handler_));
   }
 };
 
 TEST_F(GetUrlsTest, RUN_SUCCESS) {
-  MessageSharedPtr command_msg(
-      CreateMessage(NsSmartDeviceLink::NsSmartObjects::SmartType_Map));
-  (*command_msg)[am::strings::msg_params][am::strings::number] = "123";
-  (*command_msg)[am::strings::params][am::strings::connection_key] =
-      kConnectionKey;
-  (*command_msg)[am::strings::msg_params][am::hmi_request::service] =
-      kInitialService;
+  EXPECT_CALL(mock_app_manager_, GetPolicyHandler()).Times(2);
+  EXPECT_CALL(mock_policy_handler_, GetServiceUrls(kInitialService, _));
+  EXPECT_CALL(mock_policy_handler_, PolicyEnabled()).WillOnce(Return(true));
 
-  ON_CALL(mock_app_manager_, event_dispatcher())
-      .WillByDefault(ReturnRef(mock_event_dispatcher_));
+  request_command_->Run();
+}
 
-  RequestFromHMIPtr command(CreateCommand<GetUrls>(command_msg));
+TEST_F(GetUrlsTest, RUN_PolicyNotEnabled_UNSUCCESS) {
+  EXPECT_CALL(mock_app_manager_, GetPolicyHandler());
+  EXPECT_CALL(mock_policy_handler_, PolicyEnabled()).WillOnce(Return(false));
+  EXPECT_CALL(mock_app_manager_, ManageHMICommand(command_msg_))
+      .WillOnce(Return(true));
 
-  EXPECT_CALL(mock_app_manager_, GetPolicyHandler()).Times(2);
+  request_command_->Run();
+
+  EXPECT_EQ((*command_msg_)[strings::params][strings::message_type].asInt(),
+            am::MessageType::kResponse);
+  EXPECT_EQ((*command_msg_)[strings::params][am::hmi_response::code].asInt(),
+            Common_Result::DATA_NOT_AVAILABLE);
```

Comment:
@dcherniev, first argument should be expected result, second - actual one.
Please, check everywhere